tests/integration/common/transaction_utils.rs

```diff
@@ -614,6 +647,66 @@ pub fn to_internal_invoke_v3_tx(tx: &InvokeTransactionV3) -> InternalTransaction
     };
 }
 
+/// Convert a InvokeTransactionV3 to a SNOS InternalTransaction
+pub fn to_internal_deploy_v3_tx(tx: &DeployAccountTransactionV3) -> InternalTransaction {
+    let signature = Some(tx.signature.0.iter().map(|x| to_felt252(x)).collect());
+    let entry_point_selector = Some(Felt252::ZERO);
+    let calldata: Vec<_> = tx.constructor_calldata.0.iter().map(|x| to_felt252(x.into())).collect();
+
+    let nonce = felt_api2vm(tx.nonce.0);
+    let tip = felt_api2vm(tx.tip.0.into());
+
+    let nonce_data_availability_mode = Felt252::from(tx.nonce_data_availability_mode as u64);
+    let fee_data_availability_mode = Felt252::from(tx.fee_data_availability_mode as u64);
+    let resource_bounds = &tx.resource_bounds;
+
+    let paymaster_data: Vec<Felt252> = tx.paymaster_data.0.iter().map(|x| to_felt252(x.into())).collect();
+
+    let contract_address = calculate_contract_address(
+        tx.contract_address_salt,
+        tx.class_hash,
+        &tx.constructor_calldata,
+        ContractAddress::from(0_u8),
+    )
+    .unwrap();
+    let contract_address_salt = felt_api2vm(tx.contract_address_salt.0);
+    let contract_address = felt_api2vm(*contract_address.0);
```

Comment:
Nitpick but it's weird that `contract_address_salt` is converted right in the middle of generating `contract_address`.